homeassistant/components/netatmo/data_handler.py

```diff
@@ -0,0 +1,166 @@
+    HOMECOACH_DATA_CLASS_NAME: 300,
+    PUBLICDATA_DATA_CLASS_NAME: 600,
+}
+SCAN_INTERVAL = 60
+
+
+class NetatmoDataHandler:
+    """Manages the Netatmo data handling."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry):
+        """Initialize self."""
+        self.hass = hass
+        self._auth = hass.data[DOMAIN][entry.entry_id][AUTH]
+        self.listeners: List[CALLBACK_TYPE] = []
+        self._data_classes: Dict = {}
+        self.data = {}
+        self._queue: Deque = deque()
+        self._webhook: bool = False
+
+    async def async_setup(self):
+        """Set up the Netatmo data handler."""
+
+        async_track_time_interval(
+            self.hass, self.async_update, timedelta(seconds=SCAN_INTERVAL)
+        )
+
+        self.listeners.append(
+            self.hass.bus.async_listen("netatmo_event", self.handle_event)
+        )
+
+    async def async_update(self, event_time):
+        """
+        Update device.
+
+        We do up to BATCH_SIZE calls in one update in order
+        to minimize the calls on the api service.
+        """
+        for data_class in islice(self._queue, 0, BATCH_SIZE):
+            if data_class[NEXT_SCAN] > time():
+                continue
+            self._data_classes[data_class["name"]][NEXT_SCAN] = (
+                time() + data_class["interval"]
+            )
+
+            await self.async_fetch_data(
+                data_class["class"], data_class["name"], **data_class["kwargs"]
+            )
+
+        self._queue.rotate(BATCH_SIZE)
+
+    async def async_cleanup(self):
+        """Clean up the Netatmo data handler."""
+        for listener in self.listeners:
+            listener()
+
+    async def handle_event(self, event):
+        """Handle webhook events."""
+        if event.data["data"]["push_type"] == "webhook_activation":
+            _LOGGER.info("%s webhook successfully registered", MANUFACTURER)
+            self._webhook = True
+
+        elif event.data["data"]["push_type"] == "NACamera-connection":
+            _LOGGER.debug("%s camera reconnected", MANUFACTURER)
+            self._data_classes[CAMERA_DATA_CLASS_NAME][NEXT_SCAN] = time()
+
+    async def async_fetch_data(self, data_class, data_class_entry, **kwargs):
+        """Fetch data and notify."""
+        try:
+            self.data[data_class_entry] = await self.hass.async_add_executor_job(
+                partial(data_class, **kwargs), self._auth,
+            )
+            for update_callback in self._data_classes[data_class_entry][
+                "subscriptions"
+            ]:
+                if update_callback:
+                    update_callback()
+
+        except (pyatmo.NoDevice, pyatmo.ApiError) as err:
+            _LOGGER.debug(err)
+
+    async def register_data_class(
+        self, data_class_name, data_class_entry, update_callback, **kwargs
+    ):
+        """Register data class."""
+        if data_class_entry not in self._data_classes:
```

Comment:
We can invert this check and append and return if true. Then we can outdent below.